File: dart/lib/src/v1/models/message_model.dart
import 'message_priority.dart';

class MessageModel {
  String number;
  String message;
  String senderId;
  MessagePriority priority;

  MessageModel({
    required this.number,
    required this.message,
    this.senderId = "EgoSms",
    this.priority = MessagePriority.HIGHEST,
  });

  Map<String, dynamic> toJson() => {
    'number': number,
    'message': message,
    'senderid': senderId,
    'priority': priority.index.toString(),
  };
}

File: c#/src/v1/models/MessageModel.cs
using System;
using System.Text.Json.Serialization;

namespace Comms.Models;

public class MessageModel
{
    [JsonPropertyName("number")]
    public string Number { get; set; }
    [JsonPropertyName("message")]
    public string Message { get; set; }
    [JsonPropertyName("senderid")]
    public string SenderId { get; set; }
    [JsonPropertyName("priority")]
    public MessagePriority Priority { get; set; } = MessagePriority.Highest;

    public MessageModel(string number, string message, string senderId, MessagePriority priority)
    {
        Number = number ?? throw new ArgumentNullException(nameof(number));
        Message = message ?? throw new ArgumentNullException(nameof(message));
        SenderId = senderId ?? "Comms";
        Priority = priority;
    }
}
